File: weinc_src/front-admin-main/src/models/permission/permission.ts
declare module 'PermissionInfoModule' {
  export interface AvailablePermission {
    category: string | null;
    code: string;
    description: string;
    group: string | null;
    name: string;
    type: string;
  }
  export interface Permission {
    permission_code: string;
    name: string;
    target: string;
    exclude: string | null;
  }

  export interface Class {
    class_code: string;
    permissions: Permission[];
  }

  export interface PermissionInfo {
    Class: Class[];
    Member: Permission[];
  }
}

File: weinc_src/backend-admin-main/app/models/permission.py
from pydantic import BaseModel
from typing import List, Dict, Optional


class DataPermission(BaseModel):
    code: str
    name: str
    description: Optional[str]
    type: Optional[str]
    category: Optional[str]
    group: Optional[str]
    exclude: Optional[str]

    class Config:
        orm_mode = True


class DataMemberPermission(BaseModel):
    permissions: List[Dict[str, str]]
